src/main/java/dylanbot/Storage.java

```diff
@@ -0,0 +1,110 @@
+package dylanbot;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.regex.Pattern;
+
+// deals with loading tasks from the file and saving tasks in the file
+
+/**
+ * Represents a Storage object that handles loading and saving of tasks to/from a file
+ */
+public class Storage {
+    private final String filePath;
+
+    private final Ui ui;
+
+    /**
+     * Creates a Storage object that reads/writes to the file at the specified file path
+     *
+     * @param filePath The file path of the desired file
+     * @param ui The Ui to be used
+     */
+    public Storage(String filePath, Ui ui) {
+        this.filePath = filePath;
+        this.ui = ui;
+    }
+
+    /**
+     * Loads and processes data from a specified file, returning it as an ArrayList of Tasks
+     *
+     * @return ArrayList of Tasks
+     * @throws IOException If file cannot be found at the specified file path
+     */
+    public ArrayList<Task> loadDataFromFile() throws IOException {
+        BufferedReader reader = new BufferedReader(new FileReader(filePath));
+        String nextLine;
+        ArrayList<Task> res = new ArrayList<>();
+        try {
+            while ((nextLine = reader.readLine()) != null) {
```

Comment:
Nice implementation! Perhaps you could extract some of this code and put in Parser class instead?